docs/self-managed/platform-deployment/helm-kubernetes/platforms/redhat-openshift.md

```diff
@@ -281,34 +281,49 @@ If you should decide to use the Red Hat endorsed [NGINX Ingress Controller](http
 
 ### Required steps
 
 1. Provide [TLS secrets](https://kubernetes.io/docs/concepts/configuration/secret/#tls-secrets) for the Zeebe Gateway, the [Cert Manager](https://docs.openshift.com/container-platform/4.11/security/cert_manager_operator/index.html) might be helpful here:
 
-- One issued to the Zeebe Gateway Service Name. This must use the [pkcs8 syntax](https://www.openssl.org/docs/man3.1/man1/openssl-pkcs8.html) as Zeebe only supports this, referenced as **Service Certificate Secret** or `<SERVICE_CERTIFICATE_SECRET_NAME>`. For more details, review the [OpenShift documentation](https://docs.openshift.com/container-platform/4.11/networking/routes/secured-routes.html#nw-ingress-creating-a-reencrypt-route-with-a-custom-certificate_secured-routes).
+- One issued to the Zeebe Gateway Service Name. This must use the [pkcs8 syntax](https://www.openssl.org/docs/man3.1/man1/openssl-pkcs8.html) or [pkcs1 syntax](https://en.wikipedia.org/wiki/PKCS_1) as Zeebe only supports these, referenced as **Service Certificate Secret** or `<SERVICE_CERTIFICATE_SECRET_NAME>`. For more details, review the [OpenShift documentation](https://docs.openshift.com/container-platform/4.11/networking/routes/secured-routes.html#nw-ingress-creating-a-reencrypt-route-with-a-custom-certificate_secured-routes).
 - One that is used on the exposed route, referenced as **External URL Certificate Secret** or `<EXTERNAL_URL_CERTIFICATE_SECRET_NAME>`.
 
 1. Configure your Zeebe Gateway Ingress to create a [re-encrypt route](https://docs.openshift.com/container-platform/4.11/networking/routes/route-configuration.html#nw-ingress-creating-a-route-via-an-ingress_route-configuration):
 
 ```yaml
-zeebe-gateway:
+zeebeGateway:
   ingress:
-    annotations:
-      route.openshift.io/termination: reencrypt
-      route.openshift.io/destination-ca-certificate-secret: <SERVICE_CERTIFICATE_SECRET_NAME>
-    className: openshift-default
-    tls:
-      enabled: true
-      secretName: <EXTERNAL_URL_CERTIFICATE_SECRET_NAME>
+    grpc:
+      annotations:
+        route.openshift.io/termination: reencrypt
+        route.openshift.io/destination-ca-certificate-secret: <SERVICE_CERTIFICATE_SECRET_NAME>
+      className: openshift-default
+      tls:
+        enabled: true
+        secretName: <EXTERNAL_URL_CERTIFICATE_SECRET_NAME>
+    rest:
+      annotations:
+        route.openshift.io/termination: reencrypt
+        route.openshift.io/destination-ca-certificate-secret: <SERVICE_CERTIFICATE_SECRET_NAME>
+      className: openshift-default
+      tls:
+        enabled: true
+        secretName: <EXTERNAL_URL_CERTIFICATE_SECRET_NAME>
 ```
 
 3. Mount the **Service Certificate Secret** to the Zeebe Gateway Pod:
 
 ```yaml
-zeebe-gateway:
+zeebeGateway:
   env:
+    - name: SERVER_SSL_ENABLED
+      value: "true"
+    - name: SERVER_SSL_CERTIFICATE
+      value: /usr/local/zeebe/config/tls.crt
+    - name: SERVER_SSL_CERTIFICATEPRIVATEKEY
+      value: /usr/local/zeebe/config/tls.key
```

Comment:
:thought_balloon: These new env vars are required to secure the REST API for Zeebe. See: https://stage.docs.camunda.io/docs/next/self-managed/zeebe-deployment/configuration/gateway-config/#server

Reply:
I would revert these changes as well.

When using http2/grpc, Openshift is not able to termine tls on the ingress controller (haproxy), so a re-encrypt route can be used as shown here.

This problem does not exist for rest.